screen-loans/src/main/java/com/ivy/loans/loandetails/LoanDetailsScreenState.kt

```diff
@@ -0,0 +1,22 @@
+package com.ivy.loans.loandetails
+
+import com.ivy.legacy.datamodel.Account
+import com.ivy.legacy.datamodel.Loan
+import com.ivy.loans.loan.data.DisplayLoanRecord
+import com.ivy.wallet.ui.theme.modal.LoanModalData
+import com.ivy.wallet.ui.theme.modal.LoanRecordModalData
+
+data class LoanDetailsScreenState(
+    val baseCurrency: String = "",
+    val loan: Loan? = null,
+    val displayLoanRecords: List<DisplayLoanRecord> = emptyList(),
```

Comment:
In the Compose UI state, we must use `ImmutableList` for performance reasons. Lint will complain about that, too